packages/flutter_tools/lib/src/migrations/swift_package_manager_integration_migration.dart

```diff
@@ -417,14 +430,39 @@ $newContent
     }
   }
 
-  void _ensureNewIdentifiersNotUsed(String originalProjectContents) {
-    if (originalProjectContents.contains(_flutterPluginsSwiftPackageBuildFileIdentifier)) {
+  /// Check if the reserved identifiers are used outside of the allowed sections in the pbxproj file.
+  ///
+  /// When the migrator is rerun, it should not fail
+  /// when encountering the reserved identifiers in existing, migrated sections of the file.
+  ///
+  /// For example:
+  /// - An unmigrated project is migrated on the first try
+  /// - A new application target is added to the project, through XCode
+  /// - The migrator is rerun through the flutter CLI upon executing `flutter run`
+  /// - The migrator should skip sections like 'PBXBuildFile' or 'XCSwiftPackageProductDependency',
+  ///   since they are already migrated
+  void _ensureNewIdentifiersNotUsed(List<String> lines) {
```

Comment:
When adding the missing check for already migrated targets back, I encountered an issue:
The `_ensureNewIdentifiersNotUsed` is too conservative with its checks.

So I had to fix this by allowing the identifiers in the sections where these might occur _after_ the migrator ran once. This enables the migrator to run again when additional application targets are added.
I'm not sure if there is a better solution to this issue?